services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java

```diff
@@ -1474,7 +1484,94 @@ public void sendPortUpEvents(final IOFSwitch sw) throws SwitchOperationException
+        
+        OFPortDesc ofPortDesc = getPort(sw, request.getPortNo());
+        logger.debug("Switch: {}, port: {}", sw, ofPortDesc);
+        
+        PortStatus newStatus = PortStatus.getPortStatus(request.getStatus());
+        if (newStatus != null) {
+            updatePortStatus(sw, ofPortDesc, newStatus);
+        }
+
+        if (request.getSpeed() > 0) {
+            updatePortSpeed(sw, ofPortDesc, request.getSpeed());
+        }
 
+        return new PortConfigurationResponse(request.getSwitchId(), request.getPortNo(), true);
+    }
+    
+    private void updatePortStatus(final IOFSwitch sw, final OFPortDesc ofPortDesc, final PortStatus portStatus) {
+        OFPortDesc.Builder builder = ofPortDesc.createBuilder();
+    
+        Set<OFPortConfig> configs = builder.getConfig() == null 
+                ? EnumSet.noneOf(OFPortConfig.class) : new HashSet<>(builder.getConfig());
+    
+        if (portStatus == PortStatus.UP) {
+            configs.remove(OFPortConfig.PORT_DOWN);
+        } else {
+            configs.add(OFPortConfig.PORT_DOWN);
+        }
+        builder.setConfig(configs);
+        
+        Set<OFPortConfig> portMask = EnumSet.noneOf(OFPortConfig.class);
+        portMask.add(OFPortConfig.PORT_DOWN);
+        
+        OFPortDesc updatedPort = builder.build();
+        
+        OFPortMod ofPortMod = sw.getOFFactory().buildPortMod().setConfig(configs)
+                .setPortNo(updatedPort.getPortNo())
+                .setHwAddr(updatedPort.getHwAddr())
+                .setMask(portMask)
+                .build();
+
+        logger.debug("oFPortMod request {}", ofPortMod);
+
+        boolean result = sw.write(ofPortMod);
+        logger.debug("Update port status call result: {}", result);
+    }
+
+    private void updatePortSpeed(final IOFSwitch sw, final OFPortDesc ofPortDesc, final long speed) {
+        OFPortDesc.Builder builder = ofPortDesc.createBuilder();
+
+        if (ofPortDesc.getVersion() == OFVersion.OF_11) {
+            logger.info("version {} speed update functionality not supported", ofPortDesc.getVersion());
+        } else if (ofPortDesc.getVersion().wireVersion > OFVersion.OF_11.wireVersion 
+                && ofPortDesc.getVersion().wireVersion < OFVersion.OF_14.wireVersion) {
+            builder.setCurrSpeed(speed);
+        } else {
+            List<OFPortDescProp> properties = new ArrayList<>(ofPortDesc.getProperties());
+            properties.forEach((property) -> {
+                logger.info("property {}", property);
+            });
+            builder.setProperties(properties);
+        }
+
+        OFPortDesc updatedPortDesc = builder.build();
+        OFPortStatus ofPortStatus = sw.getOFFactory().buildPortStatus()
+                .setReason(OFPortReason.MODIFY)
+                .setDesc(updatedPortDesc).build();
+        logger.debug("oFPortMod request {}", ofPortStatus);
+        
+        boolean result = sw.write(ofPortStatus);
```

Comment:
It will not work, it is not what you think. 

https://www.opennetworking.org/images/stories/downloads/sdn-resources/onf-specifications/openflow/openflow-spec-v1.3.3.pdf#page=45

Reply:
Extra code removed.